### DISCUSSION TITLE
Active slip system FCC in 2D, 3D

### DISCUSSION THREAD

**Discussion by Freya36910** [mtex-toolbox/mtex#1833] [Ask Anything]
Hello,
I have been working on Aluminum (FCC), with MTEX for finding slip systems and Schmid factor. However, recently I encountered an issue with a specific code for 3D crystal shape with slip systems(Plane and direction). I have referred to the MTEX documentation and searched for related resources online, but I haven't been able to find a satisfactory solution to my problem.  
Best Regards,
Freya

**Comment by ralfHielscher** [comment]
Feel free to post your issue here. We will check it.

Ralf.

**Reply by Freya36910** [reply]
Thanks for your response.
I found a Active slip system in 2d and 3D for BCC but I couldn't find for FCC. 

 
![3D active slip system](https://github.com/mtex-toolbox/mtex/assets/134095325/34726de9-e8d5-4a8a-8886-4a6b33e0750d)

**Reply by ralfHielscher** [reply]
Hi Fraya,

I'm afraid you need to tell us much more about your problem. What are your input data? What have you already tried to do? What exactly do you want to to? ...

So far we know way to little to give a reasonable answer.

Ralf.

**Reply by Freya36910** [reply]
Hi Ralf,

I am looking for a fatigue crack initiation in my work. I performed EBSD around the initiation point. I obtained raw data from EBSD and executed all the codes using MTEX for FCC aluminum crystal. Now, to demonstrate the incompatibility between two or three grains surrounding the fatigue initiation site, I need to determine the primary active slip system of each grain with the crystal shape in order to prove incompatibility claim.

Thanks for helping in advance
Freya

**Comment by Freya36910** [comment]
Hi Ralf,
Thanks for your reply.
I have below code it gives me 24 slip systems for FCC(aluminum), When I changed it to 'antipodal' it gives me 12 slip systems but one of grain(grain6) has different SF max which is not compatible with previous one and EBSD data channel5.
Could you please help me in this regard?
Farzaneh 

%% Schmid Factor
sS = slipSystem.fcc(ebsd.CS)
sSAll = sS.symmetrise('antipodal');
sSLocal = grains.meanOrientation * sSAll
% compute Schmid factor
Force=vector3d(1,0,0);
r=normalize(Force)
sigma = stressTensor.uniaxial(r)
SF = sSLocal.SchmidFactor(sigma);
% take the maxium allong the rows
[SFMax,active] = max(SF,[],2);
% plot the maximum Schmid factor
figure;
plot(grains,SFMax,'colorrange',[0.25 0.5])
colormap parula
c = colorbar;
c.Ticks = [0 0.1 0.2 0.3 0.4 0.5];

% take the active slip system and rotate it in specimen coordinates
sSactive = grains.meanOrientation .* sSAll(active);

% plot the maximum Schmid factor
figure;
plot(grains,SFMax,'colorrange',[0.25 0.5])
colormap parula
c = colorbar;
c.Ticks = [0 0.1 0.2 0.3 0.4 0.5];




hold on
% visualize the trace of the slip plane
quiver(grains,sSactive.trace,'color','b')

%  visualize the trace of the slip direction
quiver(grains,sSactive.b,'color','r', 'scale',50)

hold off
% *********
sSLocalMap = ebsd('Aluminium').orientations* sSAll
% compute Schmid factor
SFMap = sSLocalMap.SchmidFactor(sigma);

% take the maxium allong the rows
[SFMaxMap,active] = max(SFMap,[],2);

% plot the maximum Schmid factor
figure;
plot(ebsd('Aluminium'),SFMaxMap,'colorrange',[0.25 0.5])
colormap parula
c = colorbar;
c.Ticks = [0 0.1 0.2 0.3 0.4 0.5];

% take the active slip system and rotate it in specimen coordinates
sSactiveMap = ebsd('Aluminium').orientations .* sSAll(active);

**Comment by Freya36910** [comment]
![SF](https://github.com/mtex-toolbox/mtex/assets/134095325/7bc0beb6-1897-4853-ac81-ba2f2f8c8c3f)
